subsys/pm/pm.c

```diff
@@ -241,8 +241,8 @@ bool pm_system_suspend(int32_t ticks)
 	}
 
 #if defined(CONFIG_PM_DEVICE) && !defined(CONFIG_PM_DEVICE_RUNTIME_EXCLUSIVE)
-	if ((z_cpus_pm_state[id].state != PM_STATE_RUNTIME_IDLE) &&
-			(atomic_sub(&z_cpus_active, 1) == 1)) {
+	if ((atomic_sub(&z_cpus_active, 1) == 1) &&
+			(z_cpus_pm_state[id].state != PM_STATE_RUNTIME_IDLE)) {
```

Comment:
Is it ok that we rely here on the order of evaluation?

This is quite not obvious and IMO it's place for future issues if someone touch this code.

Reply:
agreed

Reply:
I have updated the implementation to separate out the conditions.